src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,318 @@
+   */
+  private static double cvssScoreFrom(Object configuration) {
+    // The OWASP dependency plugin is found and checking configurations.
+    for (String config : FAIL_BUILD_CONFIGURATIONS.keySet()) {
+      Optional<String> content = domParser(config, configuration);
+      String type = FAIL_BUILD_CONFIGURATIONS.get(config);
+
+      if (content.isPresent()) {
+        return parseScore(content.get(), type);
+      }
+    }
+
+    return OUT_OF_BOUND_CVSS_SCORE;
+  }
+
+  /**
+   * Parse the value depending on the given type. The type can be 'boolean' or 'number'. If
+   * 'boolean', then return 0.0, otherwise parse the given string value to 'number'.
+   * 
+   * @param value the value to be parsed.
+   * @param type the type of the string value.
+   * @return the parsed value.
+   */
+  private static double parseScore(String value, String type) {
+    if ("number".equals(type)) {
+      return Double.parseDouble(value);
+    }
+    return 0.0;
+  }
+
+  /**
+   * Get the value if a configuration exists.
+   * 
+   * @param configName The name of the configuration to find.
+   * @param configuration The list of configurations associated to OWASP Dependency check plugin.
+   * @return The value of the configuration if found.
+   */
+  private static Optional<String> domParser(String configName, Object configuration) {
+    if (configuration == null) {
+      return Optional.empty();
+    }
+
+    Xpp3Dom dom = ((Xpp3Dom) configuration).getChild(configName);
+    if (dom == null) {
+      return Optional.empty();
+    }
+
+    return Optional.ofNullable(dom.getValue());
+  }
+
+  /**
+   * Returns whether the plugin will be executed as a mandatory step during the build.
+   * 
+   * @param defaultParam The location of the plugin.
+   * @param locations The set of {@link Location}.
+   * @return True if the plugin will be executed as a mandatory step, false otherwise.
+   */
+  private static boolean mandatoryCheckFor(Set<Location> locations) {
+    return (locations.contains(BUILD) || locations.contains(REPORTING))  
+        && !locations.contains(PROFILE)
+        && !locations.contains(MANAGEMENT);
+  }
+
+  /**
+   * Creates a new visitor for searching OWASP Dependency Check in a POM file.
+   */
+  private static Visitor withVisitor() {
+    return new Visitor();
+  }
+
+  /**
+   * A visitor for searching OWASP Dependency Check in a POM file.
+   */
+  private static class Visitor implements ModelVisitor {
+
+    /**
+     * The cvss threshold configuration's cvss score.
+     */
+    private double score = OUT_OF_BOUND_CVSS_SCORE;
+
+    /**
+     * The availability status of plugin in the project.
+     */
+    private Status status = NOTFOUND;
+
+    @Override
+    public void accept(Plugin plugin, Set<Location> locations) {
+      if (isDependencyCheck(plugin)) {
+        status = availablityFrom(locations);
+        score = cvssScoreFrom(plugin.getConfiguration());
```

Comment:
Did you check if the tool can still fail the build even if it's in the reporting section? If it can't, then the provide needs to take this into account when it reads the CVSS threshold.